src/test/groovy/org/gradle/android/SimpleAndroidApp.groovy

```diff
@@ -219,26 +222,46 @@ class SimpleAndroidApp {
     }
 
     private String getKotlinPluginsIfEnabled() {
-        return kotlinEnabled ? """
+        if (kotlinEnabled && kspEnabled) {
+            return """
+             apply plugin: "kotlin-android"
+             apply plugin: "com.google.devtools.ksp"
+         """
+        } else if (kotlinEnabled)
+            return """
             apply plugin: "kotlin-android"
             apply plugin: "kotlin-kapt"
-        """ : ""
+        """ else return ""
     }
 
     private String getKotlinDependenciesIfEnabled() {
         return kotlinEnabled ? """
-            ${kaptRoomDependencyIfEnabled}
             implementation "org.jetbrains.kotlin:kotlin-stdlib"
         """ : ""
     }
 
+    private String getKspIfEnabled() {
+        return kspEnabled ? """
+         plugins {
+             id 'com.google.devtools.ksp' version '1.7.22-1.0.8'
```

Comment:
should we sync this to the kotlin version used in this project?

Reply:
absolutely, today it has been a rough day with the JDK17 versioning and Kotlin surfaced too.